presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1404,6 +1414,34 @@ void PrestoServer::reportNodeStatus(proxygen::ResponseHandler* downstream) {
   http::sendOkResponse(downstream, json(fetchNodeStatus()));
 }
 
+void PrestoServer::handleGracefulShutdown(const std::vector<std::unique_ptr<folly::IOBuf>>& body , proxygen::ResponseHandler* downstream){
+    std::string requestBody;
+    for (const auto& buf : body) {
+        // Convert IOBuf to fbstring and then to std::string
+        requestBody += buf->moveToFbString().toStdString();
+    }
+    requestBody.erase(requestBody.length()-2);
+    if (requestBody == "\"SHUTTING_DOWN\"") {
+     // Print message and initiate shutdown
+     LOG(INFO) << "Shutdown requested";
+     if (nodeState() == NodeState::kActive) {
+         // Call stop() to handle the shutdown process
+         std::thread([this]() {
+                 this->stop();
+             }).detach();
+     } else {
+         // Print message and indicate that node is inactive or shutdown is already requested
+         LOG(INFO) << "Node is inactive or shutdown is already requested";
+     }
+     http::sendOkResponse(downstream);
+    } else {
+     // Throw a 400 Bad Request error if the body has some other value
+     http::sendErrorResponse(
+         downstream,
+         "Bad Request: Body must contain 'SHUTTING_DOWN'.",http::kHttpBadRequest);
```

Comment:
I am not sure if we should say `Body must contain 'SHUTTING_DOWN'` due to security reasons.
Just `Bad Request` should do?
missing space after comma

Reply:
Ok, removing the extra text and adding the missing space.